internal/gok/overwrite.go

```diff
@@ -72,19 +74,36 @@ func (r *overwriteImplConfig) run(ctx context.Context, args []string, stdout, st
 		cfg.InternalCompatibilityFlags = &config.InternalCompatibilityFlags{}
 	}
 
+	if r.full != "" && r.gaf != "" {
+		return fmt.Errorf("cannot specify both --%s and --%s", "full", "gaf")
+	}
+
 	// gok overwrite is mutually exclusive with gok update
 	cfg.InternalCompatibilityFlags.Update = ""
 
 	// Turn all paths into absolute paths so that the output files land in the
 	// current directory despite the os.Chdir() call below.
-	for _, str := range []*string{&r.full, &r.boot, &r.root, &r.mbr} {
+	for _, str := range []*string{&r.full, &r.gaf, &r.boot, &r.root, &r.mbr} {
 		if *str != "" {
 			*str, err = filepath.Abs(*str)
 			if err != nil {
 				return err
 			}
 		}
 	}
+
+	// It's guaranteed that only one is not empty.
```

Comment:
Instead of putting the output type/path into the config, I think it would be better to implement a “Pack” struct in the internal/packer package, just like in its counter part https://pkg.go.dev/github.com/gokrazy/tools/packer#Pack

Reply:
Makes sense. Done.

Reply:
Sounds good. Left one more follow-up comment below.